### PR TITLE
C++ xDS interop test server changes

### PR DESCRIPTION
**Pull request by yashykt**

### CODE REVIEW

File: test/cpp/interop/xds_interop_server.cc [hunk 1]
```diff
@@ -94,12 +161,18 @@ int main(int argc, char** argv) {
     std::cout << "Failed to get hostname, terminating" << std::endl;
     return 1;
   }
+  int port = absl::GetFlag(FLAGS_port);
   if (absl::GetFlag(FLAGS_port) == 0) {
```
Comment (donnadionne) [inline test/cpp/interop/xds_interop_server.cc:165]: use port instead of getting flag again?

File: test/cpp/interop/xds_interop_server.cc [hunk 2]
```diff
@@ -94,12 +161,18 @@ int main(int argc, char** argv) {
...
     std::cout << "Invalid port, terminating" << std::endl;
     return 1;
   }
-
-  RunServer(absl::GetFlag(FLAGS_port), hostname);
+  int maintenance_port = absl::GetFlag(FLAGS_maintenance_port);
+  if (absl::GetFlag(FLAGS_port) == 0) {
```
Comment (donnadionne) [inline test/cpp/interop/xds_interop_server.cc:170]: use port above

Comment (donnadionne) [reply test/cpp/interop/xds_interop_server.cc:170]: actually you mean maintenance_port right?

File: test/cpp/interop/xds_interop_server.cc (hunk 1 above)
Comment (yashykt) [reply test/cpp/interop/xds_interop_server.cc]: Done!

File: test/cpp/interop/xds_interop_server.cc (hunk 2 above)
Comment (yashykt) [reply test/cpp/interop/xds_interop_server.cc]: Right, this should be `maintenance_port`

**Review by donnadionne — APPROVED** [review]

**Pull request merged**